app/presenters/dromedary/index_presenter.rb

```diff
@@ -28,6 +28,62 @@ def initialize(document, view_context, configuration = view_context.blacklight_c
       @search_field = view_context.search_state.params_for_search['search_field']
     end
 
+    ##### XSLT TRANSFORMS #####
+
+    def form_xslt
+      @form_xsl ||= Nokogiri::XSLT(File.read('./indexer/xslt/FormOnly.xsl'))
+    end
+
+    def def_xslt
+      @def_xslt ||= Nokogiri::XSLT(File.read('./indexer/xslt/DefOnly.xsl')
+    end
+
+    def cit_xslt
+      @def_xslt ||= Nokogiri::XSLT(File.read('./indexer/xslt/CitOnly.xsl')
+    end
+
+    def etym_xslt
+      @def_xslt ||= Nokogiri::XSLT(File.read('./indexer/xslt/EtymOnly.xsl')
+    end
+
+    def note_xslt
+      @def_xslt ||= Nokogiri::XSLT(File.read('./indexer/xslt/NoteOnly.xsl')
+    end
+
+    def supp_xslt
+      @def_xslt ||= Nokogiri::XSLT(File.read('./indexer/xslt/SupplementOnly.xsl')
+    end
+
+    ####### Getting nokogiri nodes #####
+
+
+    # @return [Nokogiri::Node | nil] nil if there isn't a FORM section, the node otherwise
+    def form_node
+      @nokonode.xpath('/MED/ENTRYFREE/FORM').first
+    end
+
+    # There's only one FORM section, so just take care of it here
+    # @return [String] the transformed form
+    def form_html
+      form_node.nil? ? "" : form_xslt.apply_to(form_node)
+    end
+
+    # @param [MiddleEnglishDictionary::Entry::Sense] sense The sense whose def you want
+    # @return [String] The definition transformed into HTML
+    def def_html(sense)
+      nokonode = Nokogiri::XML(sense.definition_xml)
+      nokonode.nil? ? "" : def_xslt.apply_to(nokonode)
+    end
+
+    # @param [MiddleEnglishDictionary::Entry::Sense] sense The sense whose def you want
+    # @return [String] The definition transformed into HTML
+    def supp_html(sense)      
+      nokonode = Nokogiri::XML(sense.egs_xml)
```

Comment:
Supplements only exist at the entry level; you get a (usually empty) list of them with `entry.supplements`, which you'll loop over in the view.

```ruby
def supplement_node(supp)
  Nokogiri::XML(supp.xml)
end

def supplement_html(supp)
  supp_xslt.apply_to(supplement_node)
end

```
 
And then in the view something like
```ruby
<% entry.supplements.each do |supp| %>
  <div class="if needed">
    <%== presenter.supplement_html(supp)%>
  </div>
<% end %>